addons/hr_expense/views/hr_expense_views.xml

```diff
@@ -385,19 +413,21 @@
             <field name="arch" type="xml">
                 <form string="Expense Reports" class="o_expense_sheet">
                  <header>
-                    <button name="approve_expense_sheets" states="submit" string="Approve" type="object" groups="hr_expense.group_hr_expense_user" class="oe_highlight o_expense_sheet_approve"/>
+                    <field name="is_user_the_expense_responsible" invisible="1"/>
+                    <button name="approve_expense_sheets" string="Approve" type="object" attrs="{'invisible': ['|', ('state', '!=', 'submit'), ('is_user_the_expense_responsible', '=',False)]}" class="oe_highlight o_expense_sheet_approve"/>
```

Comment:
Generally we avoid hiding buttons for things like "I am not the responsible". It should be based on access rights as on all main applications. What do you think about that change ?